Increasing top/bottom entry padding to fix its look on Eclipse

pedrosans opened the pull request:
Hi, I increased the padding on the entry so it will look better on Eclipse, sending a picture of the change impact for recefence.

![eclipse-entry](https://user-images.githubusercontent.com/509641/34855962-ceae7666-f729-11e7-8578-46ef25af74d4.png)

fossfreedom commented:
@pedrosans do you have the same picture but before the change? Need to compare and contrast.

pedrosans commented:
Hi @fossfreedom , I added such picture on the pull request message but didn't made it clear. The left side of the picture shows the Entry (with the 'Quick Access' text as placeholder) widget with the increased padding, and on the right side of the picture its the same Entry but with the original padding.

fossfreedom commented:
Thanks.

Since the change is in the common module it will change the look of every gtk3 app with entry fields. Will need to compare other apps as well to see if the look is enhanced with this change.

NicoHood commented:
I dont see the difference. Where should I look at? Can you mark it for me in red? Sorry :S

fossfreedom commented:
Happy to merge it - but will need to ensure the unity changes are tested against Ubuntu 18.04 i.e. GTK+3.28 - in the arc source this will be the GTK3.20 folder.

NicoHood commented:
@fossfreedom can we merge this one?

fossfreedom commented:
Not really. The fix is for gtk 3.18 but not in 3.20 and later

NicoHood commented:
I will postpone it then for the next release.

pedrosans commented:
Hi, I cleaned the branch to have only the proposed change. Also updated the propose so it will have less impact on the current total height of the Entry widget.

Problem: the Entry widget in Eclipse using Arc theme, and also in any application setting a low 'min-height' value for their text fields/Entry, will have a really small Entry due its total height being calculated from the formula: max(min-height, font-height) + upper-padding + lower-padding. Even for common font sizes like 11 pt (the first image in this pull request) the Entry widget will be small.

Some context: it looks like Eclipse choose to set a low 'min-height' value for Entry widgets to prevent it from have a huge size on the screen [1], which should be OK; and it is OK on Adwaita and Ambiance, two popular themes on Ubuntu. Ambiance, for instance, just sets a generous padding for the Entry and makes no restriction on the minimum size:
...
entry{
    background-image: -gtk-gradient (linear, left top, left bottom,
                                     from (shade (@base_color, 0.96)),
                                     to (@base_color));
    padding: 5px 7px;
    color: @text_color;
    text-shadow: none;
}
...

Proposal: while Arc theme strives to have a decent height for Entry widgets, it can archive the same final height but by replacing some of its pixels from the min-height to the padding itself. I tested the theme using Ubuntu Regular font with 11pt, and entries on Gnome Tweak Tool, Nautilus, Synaptic, Eclipse, Settings, they all look to have consistent heights with this proposal. The proposals keeps the min-height with a high value, preventing apps and setups with small fonts from having an Entry too small; plus adds a bigger padding also preventing small Entries and causing a good separation from the text and the Entry border, even in bigger font setups.

1 - https://bugs.eclipse.org/bugs/show_bug.cgi?id=527729
